### ISSUE TITLE
Require new package updates everything

### ISSUE THREAD

**Issue by OndrejPiza** [composer/composer#6978]
My `composer.json`:

{
    "require": {
        "symfony/finder": "^3.2",
        "symfony/dependency-injection": "^3.2",
        "symfony/class-loader": "^3.2",
        "symfony/config": "^3.2",
        "symfony/expression-language": "^3.2",
        "doctrine/orm": "^2.5",
        "defuse/php-encryption": "^2.1",
        "symfony/serializer": "^3.3",
        "symfony/property-access": "^3.3",
        "twig/twig": "~2.0",
        "jms/serializer-bundle": "^2.0",
        "guzzlehttp/guzzle": "^6.3",
        "knplabs/knp-snappy": "^1.0",
        "adldap/adldap": "^5.0"
    },
    "minimum-stability": "dev"
}

Output of `composer diagnose`:

Checking composer version: OK
Composer version: 1.6.1
PHP version: 7.0.22-0ubuntu0.16.04.1
PHP binary path: /usr/bin/php7.0

When I run this command:

composer require adldap/adldap ^5.0

I get the following output:

Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 4 updates, 0 removals
  - Updating symfony/http-kernel 3.2.x-dev (18ec42e => 493f0db):  Checking out 493f0dbc9d
...... and so on

And I expected this to happen:

I need to just adldap pacakge to be downloaded and installed, I checked it dependencies, no need to update other packages. Require should do just exactly this. Why is not doing that? Do I need to add some flag?

**Comment by Seldaek** [comment]
Most likely it's doing that because you have a lock file that is newer than your vendor directory, and you never ran `composer install` to apply it, so it does the dependency update for the new package you required and then makes sure the vendor dir is synced with the lock file.. Try running composer install before then if require still changes other packages please give more details so we can try to reproduce it.

**Comment by OndrejPiza** [comment]
Aaah, thank you. I wasn't aware that composer does some check on vendor directory against lock file. I thought that he does only check downloaded info about packages versions against lock file. Yes, the lock file is shared against team so other can change it and it is possible that there were new versions of packages which I had not installed because I did not need them right now. 

I've tried solution you've suggested and it worked.

Thanks again.
